### PR TITLE
chore(workflow): set npm registry tag

### PR DESCRIPTION
**Pull request by JAdshead**
## Description

create new action to set the dist tag for npm registy.

## Motivation and Context

to fix incorrect tags in the npm registry

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] My changes are in sync with the code style of this project.
- [ ] There aren't any other open Pull Requests for the same issue/update.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] This change impacts caching for client browsers.
- [ ] This change adds additional environment variable requirements for Holocron users.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: .github/workflows/set-tag.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        required: true
+        type: string
+jobs:
+  set-tag:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 18
+    - run: echo "npm dist-tag add $${{ github.event.inputs.name }}@$${{ github.event.inputs.version}} latest"
```
Comment (JAdshead) [inline .github/workflows/set-tag.yml:27]: I want to run with echo to ensure that this is formatted as expected

**Review by code-forger — DISMISSED** [review]

File: .github/workflows/set-tag.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: Set Npm Registry Latest Tag
```
Comment (PixnBits) [inline .github/workflows/set-tag.yml:1]: #nitpick
```suggestion
name: Set npm Registry Latest Tag
```

File: .github/workflows/set-tag.yml (hunk 1 above)
Comment (PixnBits) [inline .github/workflows/set-tag.yml:27]: I think that should be
```suggestion
    - run: echo "npm dist-tag add ${{ github.event.inputs.name }}@${{ github.event.inputs.version}} latest"
```

but probably safer to
```suggestion
    - run: echo "npm dist-tag add \"${{ github.event.inputs.name }}@${{ github.event.inputs.version}}\" latest"
```

File: .github/workflows/set-tag.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      name:
+        description: 'Name of package'
+        default: 'holocron'
+        required: true
+        type: string
+        options:
+        - holocron
+        - holocron-module-register-webpack-plugin
+        - holocron-module-route
+        - iguazu-holocron
```
Comment (Matthew-Mallimo) [inline .github/workflows/set-tag.yml:15]: type is string but we provide options?
`type: choice`

Comment (JAdshead) [reply .github/workflows/set-tag.yml:15]: is the value not a string ?

Comment (JAdshead) [reply .github/workflows/set-tag.yml:15]: the types can only be one of boolean, number, or string.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_callinputsinput_idtype

**Review by Matthew-Mallimo — APPROVED** [review]

**Review by PixnBits — APPROVED** [review]

**Comment by PixnBits** [comment]
later maybe we can add an input for the tag to update \m/

**Pull request merged**